test/test_common/environment.cc

```diff
@@ -41,8 +41,8 @@ std::string makeTempDir(char* name_template) {
   char* dirname = ::_mktemp(name_template);
   RELEASE_ASSERT(dirname != nullptr, fmt::format("failed to create tempdir from template: {} {}",
                                                  name_template, strerror(errno)));
-#ifdef __cpp_lib_filesystem
-  std::filesystem::create_directories(dirname);
+#if defined(_LIBCPP_VERSION) && TEST_STD_VER < 17 && !defined(__APPLE__)
+  std::__fs::filesystem::create_directories(dirname);
```

Comment:
What happens if someone is building under a C++17 compiler option? Do they just fallback to `mkdir` rather than the `std::filesystem` implementation? Should Envoy build cleanly under C++17 (even if we are only using C++14 features)?

Reply:
Hmm. Actually, that branch should be good for C++17, they shouldn't fall back to mkdir. Will update / recheck.

http://clang-developers.42468.n3.nabble.com/Filesystem-has-Landed-in-Libc-td4061350.html